sel/src/Sel/Hashing/Generic.hs

```diff
@@ -64,6 +66,24 @@ import LibSodium.Bindings.GenericHashing (cryptoGenericHash, cryptoGenericHashBy
 -- @since 0.0.1.0
 newtype HashKey = HashKey (ForeignPtr CUChar)
 
+instance Eq HashKey where
+ (HashKey pk1) == (HashKey pk2) = 
+    let p = unsafeForeignPtrToPtr pk1
+        q = unsafeForeignPtrToPtr pk2
+     in unsafeDupablePerformIO $
+          do p' <- peek p
+             q' <- peek q
```

Comment:
This will not work - you are comparing one `CUChar`, but the `HashKey` is longer. You need to use something like `memcmp` or read off in a loop.

Reply:
Would perhaps https://hackage.haskell.org/package/bytestring-0.11.4.0/docs/Data-ByteString-Internal.html#v:fromForeignPtr work? After all CUChar is a newtype over Word8 right?

Reply:
Or would the overhead be too big and indeed, byte-by-byte comparison without creating a bytestring is better?

Reply:
Creating a `ByteString` requires _two_ additional `Int`s (length and offset). Thus, this strategy would carry an overhead of 64 bytes, which is a bit ridiculous considering the size of what's being compared.

Seriously, just use `memcmp`, it's not that hard.